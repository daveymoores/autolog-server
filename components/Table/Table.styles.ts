import css from "styled-jsx/css";

export default css`
  .timesheet__table .timesheet__table--wrapper {
    display: flex;
    flex-direction: column;
  }

  :global(.timesheet__table .timesheet__cell--title:nth-child(1)) {
    align-self: flex-start;
    border-bottom: none;
  }

  :global(.timesheet__table .timesheet__cell--title:last-child) {
    align-self: flex-end;
  }
`;
