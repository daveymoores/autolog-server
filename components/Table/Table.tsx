import React from "react";

import { ParsedTimesheetDayLog, TimesheetProps } from "../../pages/[timesheet]";
import Cell from "../Cell/Cell";
import TableContainer from "../TableContainer/TableContainer";
import styles from "./Table.styles";

type TableProps = Pick<
  TimesheetProps<ParsedTimesheetDayLog>,
  "timesheet_log" | "total_hours"
>;

const Table: React.FC<TableProps> = ({ timesheet_log, total_hours }) => {
  return (
    <>
      <div className="timesheet__table">
        <div className="timesheet__table--wrapper">
          <Cell text={`Days:`} title />
          <TableContainer timesheet_log={timesheet_log} />
          <Cell text={`Total hours worked: ${total_hours}`} title />
        </div>
      </div>
      <style jsx>
        {`
          .timesheet__table {
            max-width: calc(
              ${timesheet_log.length} * var(--cellHeight) +
                ${timesheet_log.length} * var(--lineWidth)
            );
            margin: auto;
          }
        `}
      </style>
      <style jsx>{styles}</style>
    </>
  );
};

export default Table;
