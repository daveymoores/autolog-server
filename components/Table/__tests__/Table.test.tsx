import React from "react";
import renderer from "react-test-renderer";

import Table from "../Table";

describe("Table", () => {
  it("renders Table unchanged", () => {
    const tree = renderer
      .create(
        <Table
          timesheet_log={[{ user_edited: false, hours: 0, weekend: false }]}
          total_hours={48}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
