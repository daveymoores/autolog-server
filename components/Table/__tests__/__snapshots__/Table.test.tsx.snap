// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders Table unchanged 1`] = `
<div
  className="mt-2"
>
  <div
    className="flex flex-col"
  >
    <div
      className="relative"
    >
      <div
        className="lg:w-full overflow-scroll lg:overflow-auto bg-slate-900 py-4 lg:bg-transparent lg:py-0 lg:after:hidden after:bg-gradient-to-r after:from-slate-900 after:to-transparent after:bg-transparent after:content-[''] after:absolute after:h-full after:w-6 sm:after:w-8 after:top-0 lg:before:hidden before:bg-gradient-to-l before:from-slate-900 before:to-transparent before:bg-transparent before:content-[''] before:absolute before:h-full before:w-6 sm:before:w-8 before:top-0 before:right-0"
      >
        <div
          className="w-[1200px] container max-w-none lg:w-full lg:max-w-screen-lg xl:max-w-screen-xl 2xl:max-w-screen-2xl"
        >
          <div>
            <div />
          </div>
          <div>
            <div
              className="text-sm lg:text-md text-center font-semibold"
            >
              1
            </div>
          </div>
          <div>
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full container flex flex-col"
    >
      <p
        className="font-semibold text-md text-slate-500 mt-6 self-end"
      >
        foobar
         total:
         
        <span
          className="text-green-100 font-bold"
        >
          48
           hours
        </span>
      </p>
    </div>
  </div>
</div>
`;
